packages/client/hmi-client/src/services/calibrate-workflow.ts

```diff
@@ -116,27 +117,42 @@ export const renderLossGraph = (
 	yAxisGroup.attr('transform', `translate(${marginLeft}, 0)`).call(yAxis);
 };
 
-export const autoCalibrationMapping = async (modelOptions: State[], datasetOptions: any[]) => {
+export const autoCalibrationMapping = async (
+	modelOptions: State[],
+	datasetOptions: DatasetColumn[]
+) => {
 	const result = [] as CalibrateMap[];
-	modelOptions.forEach((modelOption) => {
-		datasetOptions.forEach((datasetOption) => {
+	const acceptableDistance = 0.7;
+	for (let i = 0; i < modelOptions.length; i++) {
+		for (let j = 0; j < datasetOptions.length; j++) {
 			// Check for direct string match
-			if (modelOption.id.toLowerCase() === datasetOption.name.toLowerCase()) {
-				result.push({ modelVariable: modelOption.id, datasetVariable: datasetOption.name });
+			if (modelOptions[i].id.toLowerCase() === datasetOptions[j].name.toLowerCase()) {
+				result.push({ modelVariable: modelOptions[i].id, datasetVariable: datasetOptions[j].name });
 			}
 			// No direct string match, check grounding keys (if they exist)
 			else if (
-				modelOption.grounding?.identifiers &&
-				datasetOption?.metadata?.groundings?.identifiers
+				modelOptions[i].grounding?.identifiers &&
+				datasetOptions[j]?.metadata?.groundings?.identifiers
 			) {
-				const datasetKeys = Object.keys(datasetOption.metadata.groundings.identifiers);
-				Object.keys(modelOption.grounding.identifiers).forEach((modelGrounding) => {
-					if (datasetKeys.includes(modelGrounding)) {
-						result.push({ modelVariable: modelOption.id, datasetVariable: datasetOption.name });
+				const modelTemp = Object.entries(modelOptions[i].grounding?.identifiers);
+				const datasetTemp = Object.entries(datasetOptions[j].metadata?.groundings?.identifiers);
+				const modelGroundingList = modelTemp.map((ele) => ele.join(':'));
+				const dataGroundingList = datasetTemp.map((ele) => ele.at(0) as string);
```

Comment:
confirming this is always correct. KMW may not provide such consistent formatting